File: Data/Data.Models/Helpers/DateTimeHelper.cs
using System;

namespace Data.Models.Helpers
{
    public class DateTimeHelper
    {

        public static DateTime FromString(string s)
        {
            var arr = s.Split("-");
            return new DateTime(
                int.Parse(arr[0]),
                int.Parse(arr[1]),
                int.Parse(arr[2]),
                int.Parse(arr[3]),
                int.Parse(arr[4]),
                0
                );
        }

        public static string ToString(DateTime date)
        {
            return string.Join("-",
                date.Year,
                date.Month,
                date.Date,
                date.Hour,
                date.Minute);
        }

        public static string ExtractTime(string s)
        {
            var time = FromString(s);
            return time.Hour + ":" + time.Minute;
        }

        public static string FromDateTimeString(string dateTime)
        {
            return dateTime.Replace(":", "-").Replace("T", "-").Replace("-0", "-");
        }

    }
}

File: Logic/src/helpers/DateTimeHelper.java
package helpers;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class DateTimeHelper {

    public static boolean overlaps(
            LocalDateTime date1Start, LocalDateTime date1End,
            LocalDateTime date2Start, LocalDateTime date2End)
    {
        if (date1End.isBefore(date1Start) || date2End.isBefore(date2Start))
            throw new IllegalArgumentException("Date end has to be after date start.");

        return !(date1End.isBefore(date2Start) || date2End.isBefore(date1Start));
    }

    public static LocalDateTime fromString(String s) {
        String[] arr = s.split("-");

        return LocalDateTime.of(
                Integer.parseInt(arr[0]),
                Integer.parseInt(arr[1]),
                Integer.parseInt(arr[2]),
                Integer.parseInt(arr[3]),
                Integer.parseInt(arr[4])
        );
    }

    public static String toString(LocalDateTime t) {
        return String.join("-",
                t.getYear() + "",
                t.getMonthValue() + "",
                t.getDayOfMonth() + "",
                t.getHour() + "",
                t.getMinute() + "");
    }

    public static String getCurrentTime() {
        return toString(LocalDateTime.now());
    }
}
